saleor/core/jwt_manager.py

```diff
@@ -0,0 +1,159 @@
+            return cls._load_local_private_key(key_path)
+
+        return cls._create_local_private_key(key_path)
+
+    @classmethod
+    def _load_local_private_key(cls, path) -> "rsa.RSAPrivateKey":
+        with open(path, "rb") as key_file:
+            return serialization.load_pem_private_key(
+                key_file.read(), password=None
+            )  # type: ignore
+
+    @classmethod
+    def _create_local_private_key(cls, path) -> "rsa.RSAPrivateKey":
+        private_key = rsa.generate_private_key(
+            public_exponent=65537,
+            key_size=2048,
+        )
+        with open(path, "wb") as p_key_file:
+            p_key_file.write(
+                private_key.private_bytes(
+                    encoding=serialization.Encoding.PEM,
+                    format=serialization.PrivateFormat.TraditionalOpenSSL,
+                    encryption_algorithm=serialization.NoEncryption(),
+                )
+            )
+        return private_key
+
+    @classmethod
+    def get_public_key(cls, *_args, **_kwargs):
+        private_key = cls.get_private_key()
+        global PUBLIC_KEY
+        if PUBLIC_KEY is None:
+            PUBLIC_KEY = private_key.public_key()
+        return PUBLIC_KEY
+
+    @classmethod
+    def get_jwks(cls) -> dict:
+        jwk_dict = json.loads(RSAAlgorithm.to_jwk(cls.get_public_key()))
+        jwk_dict.update({"use": "sig", "kid": KID})
+        return {"keys": [jwk_dict]}
+
+    @classmethod
+    def encode(cls, payload):
+        return jwt.encode(
+            payload, cls.get_private_key(), algorithm="RS256", headers={"kid": KID}
+        )
+
+    @classmethod
+    def decode(cls, token, verify_expiration: bool = True):
+        headers = jwt.get_unverified_header(token)
+        if headers.get("alg") == "RS256":
+            return jwt.decode(
+                token,
+                cls.get_public_key(),
+                algorithms=["RS256"],
+                options={"verify_exp": verify_expiration},
+            )
+        return jwt.decode(
+            token,
+            settings.SECRET_KEY,  # type: ignore
+            algorithms=["HS256"],
+            options={"verify_exp": verify_expiration},
+        )
+
+    @classmethod
+    def validate_configuration(cls):
+        if not settings.RSA_PRIVATE_KEY:
+            if not settings.DEBUG:
+                raise ImproperlyConfigured(
+                    "Variable RSA_PRIVATE_KEY is not provided. "
+                    "It is required for running in not DEBUG mode."
+                )
+            else:
+                msg = (
+                    "RSA_PRIVATE_KEY is missing. Using temporary key for local "
+                    "development with DEBUG mode."
+                )
+                logger.warning(color_style().WARNING(msg))
+
+        try:
+            cls.get_private_key()
+        except Exception as e:
+            raise ImproperlyConfigured(f"Unable to load provided pem private key. {e}")
```

Comment:
I think it should be referred to as PEM in the text:
```suggestion
            raise ImproperlyConfigured(f"Unable to load provided PEM private key. {e}")
```